workspaces/marketplace/packages/cli/src/commands/export-csv/index.ts

```diff
@@ -0,0 +1,241 @@
+    const removedNewLines = cell.replace(/\n/g, '\\n');
+    const regex = /[\s,"]/;
+
+    // adapted from https://stackoverflow.com/a/47340986
+    if (regex.exec(removedNewLines.replace(/ /g, ''))) {
+      return `"${removedNewLines.replace(/"/g, '""')}"`;
+    }
+
+    return cell;
+  };
+
+  /** Add a row to the CSV */
+  addRow(row: T) {
+    this.rows.push(row);
+  }
+
+  /**
+   * @param includeHeader Whether to include the header row in the CSV
+   * @returns A CSV generated from the rows and columns provided
+   */
+  generate(includeHeader = true): string {
+    const header = this.columns.map(column => column.key).join(',');
+    const rows = this.rows
+      .map(row =>
+        this.columns
+          .map(column =>
+            CSVGenerator.escapeCSVCell(String(column.getter(row) || '')),
+          )
+          .join(','),
+      )
+      .join('\n');
+
+    return includeHeader ? `${header}\n${rows}` : rows;
+  }
+}
+
+/**
+ * Zips two arrays `keys` and `getters` into an array of CSVGenerator columns.
+ *
+ * `keys` and `getters` must have the same length.
+ *
+ * @param keys The keys to be used in the CSV header
+ * @param getters The getter functions for each cell
+ */
+const getColumns = <T = any>(keys: string[], getters: ((obj: T) => any)[]) =>
+  keys.map((key, index) => ({ key, getter: getters[index] }));
+
+/**
+ * A type guard for Entity
+ *
+ * @param entity The entity to check
+ * @returns Whether the entity is an Entity
+ */
+const isEntity = (entity: any): entity is Entity => {
+  return (
+    !!entity &&
+    typeof entity === 'object' &&
+    'apiVersion' in entity &&
+    'kind' in entity
+  );
+};
+
+/**
+ * Parse a list of YAML files into a list of entities
+ * @param yamls The list of paths of YAML files to parse
+ * @returns A list of entities
+ */
+const parseYamls = async (yamls: string[]) => {
+  return await Promise.all(
+    (yamls || []).flatMap(basePath => {
+      return fs
+        .readdirSync(basePath)
+        .filter(f => f.endsWith('.yaml'))
+        .map(async (file: string) => {
+          const fileContent = await fs.readFile(
+            path.join(basePath, file),
+            'utf-8',
+          );
+
+          // read each individual plugin from the file
+          const plugins = fileContent
+            .split('\n---\n')
+            .filter(Boolean)
+            .map((plugin: string) => {
+              return YAML.parse(plugin);
+            })
+            .filter(isEntity);
+          return plugins;
+        });
+    }),
+  ).then(plugins => plugins.flat());
+};
+
+/**
+ * Fetches a `package.json` from a given package from the npm registry.
+ * @param packageName The name of the package to fetch
+ * @returns The package.json object
+ */
+const fetchNpmPackageJson = async (packageName: string, version: string) => {
+  const response = await fetch(
+    `https://registry.npmjs.org/${packageName}/${version}`,
+  );
+
+  return await response.json();
+};
```

Comment:
This is a "nice idea" but there is no need to fetch external information. All information should be part of the Plugin or Package YAML.

Esp. the descriptions and version strings. You can drop that code.